client/src/send_and_confirm_transactions_in_parallel.rs

```diff
@@ -196,71 +250,48 @@ async fn sign_all_messages_and_send<T: Signers + ?Sized>(
-        let send_over_rpc = if let Some(tpu_client) = tpu_client {
-            !tpu_client
-                .send_wire_transaction(serialized_transaction.clone())
-                .await
-        } else {
-            true
-        };
-        if send_over_rpc {
-            if let Err(e) = rpc_client.send_transaction(&transaction).await {
-                match &e.kind {
-                    ErrorKind::Io(_) | ErrorKind::Reqwest(_) => {
-                        // fall through on io error, we will retry the transaction
-                    }
-                    ErrorKind::TransactionError(transaction_error) => {
-                        context.error_map.insert(*index, transaction_error.clone());
-                        continue;
-                    }
-                    ErrorKind::RpcError(rpc_error) => {
-                        if let RpcError::RpcResponseError {
-                            data:
-                                RpcResponseErrorData::SendTransactionPreflightFailure(simulation_result),
-                            ..
-                        } = rpc_error
-                        {
-                            if let Some(transaction_error) = &simulation_result.err {
-                                context.error_map.insert(*index, transaction_error.clone());
-                                continue;
-                            }
-                        }
-                        return Err(TpuSenderError::from(e));
-                    }
-                    _ => {
-                        return Err(TpuSenderError::from(e));
-                    }
-                }
-            }
-        }
-
         let signature = transaction.signatures[0];
-        // send to confirm the transaction
-        context.unconfirmed_transasction_map.insert(
-            signature,
-            TransactionData {
-                index: *index,
-                serialized_transaction,
-                last_valid_blockheight: blockhashdata.last_valid_blockheight,
-                message: message.clone(),
-            },
-        );
-
-        if let Some(progress_bar) = progress_bar {
-            let progress = progress_from_context_and_block_height(
+        futures.push(
+            send_transaction_with_rpc_fallback(
+                rpc_client,
+                tpu_client,
+                transaction,
+                serialized_transaction.clone(),
                 context,
-                blockhashdata.last_valid_blockheight,
-            );
-            progress.set_message_for_confirmed_transactions(
-                progress_bar,
-                &format!(
-                    "Sending {}/{} transactions",
-                    counter + 1,
-                    current_transaction_count,
-                ),
-            );
-        }
+                *index,
+                counter,
+            )
+            .and_then(move |_| async move {
```

Comment:
There is no await in this method, do we really need async move?

Reply:
Yes, `and_then` requires the continuation to be a future: https://docs.rs/futures-util/latest/futures_util/future/trait.TryFutureExt.html#method.and_then